applications/StructuralMechanicsApplication/custom_conditions/displacement_control_condition.cpp

```diff
@@ -0,0 +1,345 @@
+    : Condition(rOther) {}
+
+// Destructor
+DisplacementControlCondition::~DisplacementControlCondition() {}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+/// Assignment operator.
+DisplacementControlCondition& DisplacementControlCondition::operator=(DisplacementControlCondition const& rOther)
+{
+    //ALL MEMBER VARIABLES THAT MUST BE KEPT IN AN "=" OPERATION NEEDS TO BE COPIED HERE
+
+    Condition::operator=(rOther);
+    Flags::operator=(rOther);
+    return *this;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Condition::Pointer DisplacementControlCondition::Create(
+    IndexType NewId,
+    NodesArrayType const& rThisNodes,
+    PropertiesType::Pointer pProperties
+    ) const
+{
+    KRATOS_TRY
+    return Kratos::make_intrusive<DisplacementControlCondition>(NewId, GetGeometry().Create(rThisNodes), pProperties);
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Condition::Pointer DisplacementControlCondition::Create(
+    IndexType NewId,
+    GeometryType::Pointer pGeom,
+    PropertiesType::Pointer pProperties
+    ) const
+{
+    KRATOS_TRY
+    return Kratos::make_intrusive<DisplacementControlCondition>(NewId, pGeom, pProperties);
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Condition::Pointer DisplacementControlCondition::Clone (
+    IndexType NewId,
+    NodesArrayType const& rThisNodes
+    ) const
+{
+    KRATOS_TRY
+
+    Condition::Pointer p_new_cond = Kratos::make_intrusive<DisplacementControlCondition>(NewId, GetGeometry().Create(rThisNodes), pGetProperties());
+    p_new_cond->SetData(GetData());
+    p_new_cond->Set(Flags(*this));
+    return p_new_cond;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void DisplacementControlCondition::EquationIdVector(
+    EquationIdVectorType& rResult,
+    ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_TRY
+
+    const SizeType number_of_nodes = GetGeometry().size();
+    const SizeType block_size = GetBlockSize();
+    if (rResult.size() != number_of_nodes * block_size) {
+        rResult.resize(number_of_nodes * block_size, false);
+    }
+
+    for (SizeType i = 0; i < number_of_nodes; ++i) {
+        int index = i * 2;
+        if (GetValue(POINT_LOAD_X) != 0.0) // prescribed displacement in x direction
```

Comment:
I would not use this comparison. I would access to the whole POINT_LOAD vector and then check component by component using the epsilon value